src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorOverloadResolution.cs

```diff
@@ -97,93 +134,19 @@ public void BinaryOperatorOverloadResolution(BinaryOperatorKind kind, BoundExpre
-                bool rightHadUserDefinedCandidateFromInterfaces = false;
-                string name = OperatorFacts.BinaryOperatorNameFromOperatorKind(kind);
-                var lookedInInterfaces = PooledDictionary<TypeSymbol, bool>.GetInstance();
-
-                if (!leftHadUserDefinedCandidate)
-                {
-                    leftHadUserDefinedCandidateFromInterfaces = GetUserDefinedBinaryOperatorsFromInterfaces(kind, name,
-                        leftOperatorSourceOpt, leftSourceIsInterface, left, right, ref useSiteDiagnostics, lookedInInterfaces, resultFromInterfaces.Results);
-                }
-
-                if (!rightHadUserDefinedCandidate && leftOperatorSourceOpt != rightOperatorSourceOpt)
-                {
-                    var rightOperators = ArrayBuilder<BinaryOperatorAnalysisResult>.GetInstance();
-                    rightHadUserDefinedCandidateFromInterfaces = GetUserDefinedBinaryOperatorsFromInterfaces(kind, name, 
-                        rightOperatorSourceOpt, rightSourceIsInterface, left, right, ref useSiteDiagnostics, lookedInInterfaces, rightOperators);
-
-                    AddDistinctOperators(resultFromInterfaces.Results, rightOperators);
-                    rightOperators.Free();
-                }
-
-                Debug.Assert(resultFromNonInterfaces != null || 
-                             lookedInInterfaces.Count == 0 ||
-                             (!leftHadUserDefinedCandidate && !rightHadUserDefinedCandidate));
-
-                if (leftHadUserDefinedCandidateFromInterfaces || rightHadUserDefinedCandidateFromInterfaces)
-                {
-                    result.Clear();
-
-                    if (resultFromNonInterfaces != null)
-                    {
-                        result.Results.AddRange(resultFromNonInterfaces.Results);
-                    }
-
-                    result.Results.AddRange(resultFromInterfaces.Results);
-                    BinaryOperatorOverloadResolution(left, right, result, ref useSiteDiagnostics);
-                }
-
-                resultFromInterfaces.Free();
-                lookedInInterfaces.Free();
-            }
-
-            if (resultFromNonInterfaces != null)
-            {
-                resultFromNonInterfaces.Free();
-            }
-        }
-
-        private static bool ShouldGetUserDefinedBinaryOperatorsFromInterfaces(
-            TypeSymbol operatorSourceOpt, bool sourceIsInterface,
-            ref HashSet<DiagnosticInfo> useSiteDiagnostics)
-        {
-            return (object)operatorSourceOpt != null &&
-                   (sourceIsInterface ||
-                    (operatorSourceOpt.IsTypeParameter() &&
-                     !((TypeParameterSymbol)operatorSourceOpt).AllEffectiveInterfacesWithDefinitionUseSiteDiagnostics(ref useSiteDiagnostics).IsEmpty));
         }
 
         private bool GetUserDefinedBinaryOperatorsFromInterfaces(BinaryOperatorKind kind, string name, 
```

Comment:
Oh, I see, perhaps `CandidateOperators` performs the applicability test.

---
In reply to: [134575311](https://github.com/dotnet/roslyn/pull/21623#discussion_r134575311) [](ancestors = 134575311)